packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -1153,8 +1176,77 @@ class TextStyle with Diagnosticable {
+        if (a.background == null && b.background == null && a.backgroundColor == null && b.backgroundColor == null) 'backgroundColor',
+        if (a.fontSize == null && b.fontSize == null) 'fontSize',
+        if (a.letterSpacing == null && b.letterSpacing == null) 'letterSpacing',
+        if (a.wordSpacing == null && b.wordSpacing == null) 'wordSpacing',
+        if (a.height == null && b.height == null) 'height',
+        if (a.decorationColor == null && b.decorationColor == null) 'decorationColor',
+        if (a.decorationThickness == null && b.decorationThickness == null) 'decorationThickness',
+      ];
+      if (nullFields.isEmpty) {
+        return true;
+      }
+
+      throw FlutterError.fromParts(<DiagnosticsNode>[
+        ErrorSummary('Failed to interpolate TextStyles with different inherit values.'),
+        ErrorSpacer(),
+        ErrorDescription('The TextStyles being interpolated were:'),
+        a.toDiagnosticsNode(name: 'from', style: DiagnosticsTreeStyle.singleLine),
+        b.toDiagnosticsNode(name: 'to', style: DiagnosticsTreeStyle.singleLine),
+        ErrorDescription(
+          'The following fields are unspecified in both TextStyles:\n'
+          '${nullFields.map((String name) => '"$name"').join(', ')}.\n'
+          'When "inherit" changes during the transition, these fields may '
+          'observe abrupt value changes as a result, causing "jump"s in the '
+          'transition.'
+        ),
+        ErrorSpacer(),
+        ErrorHint(
+          'In general, TextStyle.lerp only works well when both TextStyles have '
+          'the same "inherit" value, and set roughly the same fields.',
+        ),
+        ErrorHint(
+          'If the TextStyles were directly created by you, consider bringing '
+          'them to parity to ensure a smooth transition.'
+        ),
+        ErrorSpacer(),
+        ErrorHint(
+          'If one of the TextStyles being lerped is significantly more elaborate '
+          'than the other, and has "inherited" set to false, it is often because '
+          'it is merged with another TextStyle before being lerped. Comparing '
+          'the "debugLabel"s of the two TextStyles may help identify if that was '
+          'the case.'
+        ),
+        ErrorHint(
+          'For example, you may see this error message when trying to lerp '
+          'between "ThemeData()" and "Theme.of(context)". This is because '
+          'TextStyles from "Theme.of(context)" are merged with TextStyles from '
+          'another theme and thus are more elaborate than the TextStyles from '
+          '"ThemeData()" (which is reflected in their "debugLabel"s -- '
+          'TextStyles from "Theme.of(context)" should have labels in the form of '
+          '"(<A TextStyle>).merge(<Another TextStyle>)"). The problem can be '
+          'fixed by first calling ThemeData.localize: \n'
+          'ThemeData.localize(\n'
```

Comment:
I wonder if I should mention this is just mimicking `Theme.of`. Maybe there're better examples?

Reply:
This error report is probably comprehensive enough without this additional explanation. Also: it's not clear that we really want to advocate calling ThemeData.localize, since that's done automatically when the app's locale is finalized. I think it would be best to leave this part of the error report out.

Reply:
There's no `ThemeData.merge` so the example is really awkward. Should I say instead of transitioning to `Theme.of(context)` you should transition to `isDark ? ThemeData.dark() : ThemeData()`?